src/org/thoughtcrime/securesms/preferences/ChatBackgroundActivity.java

```diff
@@ -136,13 +136,15 @@ private void scaleAndSaveImage(Context context, String destinationPath) {
             Display display = ServiceUtil.getWindowManager(context).getDefaultDisplay();
             Point size = new Point();
             display.getSize(size);
+            // resize so that the larger side fits the screen accurately
+            int largerSide = (size.x > size.y ? size.x : size.y);
             Bitmap scaledBitmap = GlideApp.with(context)
                     .asBitmap()
                     .load(imageUri)
-                    .centerCrop()
+                    .fitCenter()
                     .skipMemoryCache(true)
                     .diskCacheStrategy(DiskCacheStrategy.NONE)
-                    .submit(size.x, size.y)
+                    .submit(largerSide, largerSide)
```

Comment:
the aspect ratio is kept intact by .fitCenter()?

Reply:
@cyBerta Yes, it is. fitCenter scales the smaller side of the source to match the given dimensions and scales the larger side to match the aspect ratio.

a 100:200 landscape image on a 50:20 screen will be scaled -> 50:100.
Result on Landscape (20:50): scale (during display) down to 25:50 and cut off 5 on the edges.
Result on Portrait (50:20): no further scaling needed, display 50:20 and cut off 80.

a 200:100 portrait image on that 50:20 screen will be scaled -> 100:50.
Result on Landscape (20:50): no scaling (during display): Display 20:50 and cut off 80
Result on Portrait (50:20): scale (during display) down to 50:25 and cut off 5.

That way the image is never distorted, just cut.